### DISCUSSION TITLE
Firebase Function

### DISCUSSION THREAD

**Discussion by LanceCAbson** [RageAgainstThePixel/com.openai.unity#187] [Q&A] [answered]
As OpenAI Keys are being stolen even when they are not stored in an app, would it be possible to allow this Unity OpenAI Lib to call a Firebase Function instead of the OpenAPI directly?

This would need the Firebase Function to be developed and the interface to it from this Unity OpenAI lib.
I would only need streaming chat & images to work initially.

Thanks in Advance.

**Comment by StephenHodgson** [comment, marked as answer]
Yes that's what I do for my apps in production.

See https://github.com/RageAgainstThePixel/com.openai.unity?tab=readme-ov-file#openai-api-proxy

**Reply by LanceCAbson** [reply]
Thanks, yes I saw that was similar to what I need, but I need it for Google firebase as everything in my apps is using that not Microsoft.
I'm not a backend programmer so I'm stuck with what to put into the node.js and then how to get your excellent unity lib to talk to the Firebase function instead of direct with openAI.
Any help would be greatly appreciated, and I could add a credit in my app. 

It's so frustrating as Apple have featured my app, but I keep having to take it down.

**Reply by StephenHodgson** [reply]
DM me on Discord and let's see what we can do
